python/lsst/ap/association/packageAlerts.py

```diff
@@ -77,6 +95,53 @@ def __init__(self, **kwargs):
         self.alertSchema = alertPack.Schema.from_uri(self.config.schemaFile)
         os.makedirs(self.config.alertWriteLocation, exist_ok=True)
 
+        if self.config.doProduceAlerts:
+
+            self.password = os.getenv("AP_KAFKA_PRODUCER_PASSWORD")
+            self.username = os.getenv("AP_KAFKA_PRODUCER_USERNAME")
+            self.server = os.getenv("AP_KAFKA_SERVER")
+            self.kafka_topic = os.getenv("AP_KAFKA_TOPIC")
+            # confluent_kafka configures all of its classes with dictionaries. This one
+            # sets up the bare minimum that is needed.
+            self.kafka_config = {
+                # This is the URL to use to connect to the Kafka cluster.
+                "bootstrap.servers": self.server,
+                # These next two properties tell the Kafka client about the specific
+                # authentication and authorization protocols that should be used when
+                # connecting.
+                "security.protocol": "SASL_PLAINTEXT",
+                "sasl.mechanisms": "SCRAM-SHA-512",
+                # The sasl.username and sasl.password are passed through over
+                # SCRAM-SHA-512 auth to connect to the cluster. The username is not
+                # sensitive, but the password is (of course) a secret value which
+                # should never be committed to source code.
+                "sasl.username": self.username,
+                "sasl.password": self.password,
+                # Batch size limits the largest size of a kafka alert that can be sent.
+                # We set the batch size to 2 Mb.
+                "batch.size": 2097152,
+                "linger.ms": 5,
+            }
+
+            try:
+                from confluent_kafka import KafkaException
+                self.kafka_exception = KafkaException
```

Comment:
KafkaException doesn't seem to be visible where I need it when it is imported here and not with the initial imports. If there is a workaround for this, I'm happy to use that instead.

Reply:
Changed to Eli's recommendation, import now at the top and `confluent_kafka` is set to `None` if it can't import. Within the init, it now starts by checking `if confluent_kafka is not None` it then sets up the alert production, otherwise it skips it all.